### DISCUSSION TITLE
Extensibility of Renovate

### DISCUSSION THREAD

**Discussion by leipert** [renovatebot/renovate#23470] [Suggest an Idea]
**What would you like Renovate to be able to do?**

We use Renovate successfully now since a while, but we want to modify it a bit to fit our use-cases, for example:

1. If the status check of the MR fails, we want to leave a message and assign a set of people based on some rules (e.g. date / timezones / out of office status).
2. We want to set an MR milestone based on the current date.

**Describe the solution you'd like**
The easiest solution would be to a hook at the end of renovates actions which gives us the list of merge requests for the current project. (Or maybe for all the projects), so that we can execute logic ourselves.

I see that this is problematic when using renovate's service, but it could be a bot flag, so that you can hook into it, when you run the bot yourselves.

**Describe alternatives you've considered**
Extend renovate for all the features required, but they are not as useful to others.

**Additional context**
---

cc/ @rarkins

**Comment by rarkins** [comment]
@leipert I think some generic pre/post hooks for various things could work (e.g. pre-run, post-run, pre-MR?). Can you work out the full list of what you'd need and we try to turn it into a generic solution? We can start with it being controlled by an admin flag, i.e. each bot owner can decide whether to turn it on or off.

Would you need to configurable by the repos, rather than the bot owner? i.e. bot owner trusts and allows repo owners to configure hooks.

**Comment by HonkingGoose** [comment]
> `@leipert` I think some generic pre/post hooks for various things could work (e.g. pre-run, post-run, pre-MR?). Can you work out the full list of what you'd need and we try to turn it into a generic solution? We can start with it being controlled by an admin flag, i.e. each bot owner can decide whether to turn it on or off.

I found a new issue about hooks, maybe that would work?

- #23050

I'm converting this Issue to a Discussion, because it got stale.

**Comment by jamietanna** [comment]
Moving conversation into https://github.com/renovatebot/renovate/discussions/24893
